src/Field/CoordinateReferenceSystemCombo/CoordinateReferenceSystemCombo.jsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {AutoComplete } from 'antd';
+const Option = AutoComplete.Option;
+
+import './CoordinateReferenceSystemCombo.less';
+
+import { Logger, UrlUtil } from '../../index.js';
+
+/**
+ * Class representating a combo to choose coordinate projection system via a
+ * dropdown menu and / or autocompletion
+ *
+ * @class The CoordinateReferenceSystemCombo
+ * @extends React.Component
+ */
+class CoordinateReferenceSystemCombo extends React.Component {
+
+  /**
+   * The className added to this component.
+   * @type {String}
+   * @private
+   */
+  className = 'react-geo-crs-combo'
+
+  static propTypes = {
+    /**
+     * The className which should be added.
+     * @type {String}
+     */
+    className: PropTypes.string,
+    /**
+     * The API to query for CRS definitions
+     * default: http://epsg.io
+     * @type {String}
+     */
+    epsgIoBaseUrl: PropTypes.string,
+    /**
+     * The empty text set if no value is given / provided
+     * @type {String}
+     */
+    emptyTextPlaceholderText: PropTypes.string,
+    /**
+     * An array of predefined crs definitions habving at least value (name of
+     * CRS) and code (e.g. EPSG-code of CRS) property
+     * @type {Array}
+     */
+    predefinedCrsDefinitions: PropTypes.arrayOf(PropTypes.shape({
+      value: PropTypes.string,
+      code: PropTypes.string
+    })),
+    /**
+     * A function
+     * @type {String}
+     */
+    onSelect: PropTypes.func
+  }
+
+  static defaultProps = {
+    emptyTextPlaceholderText: 'Please select a CRS',
+    epsgIoBaseUrl: 'https://epsg.io/',
+    onSelect: () => {}
+  }
+
+  /**
+   * Create a CRS combo.
+   * @constructs CoordinateReferenceSystemCombo
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      crsDefinitions: [],
+      value: null
+    };
+
+    this.onCrsItemSelect = this.onCrsItemSelect.bind(this);
+  }
+
+  /**
+   * Fetch CRS definitions from epsg.io for given search string
+   *
+   * @param {String} searchVal The search string
+   * @param {Function} callback The callback function to be called on success.
+   */
+  fetchCrs = (searchVal, callback) => {
+    const { epsgIoBaseUrl } = this.props;
+
+    const queryParameters = {
+      format: 'json',
+      q: searchVal
+    };
+
+    fetch(`${epsgIoBaseUrl}?${UrlUtil.objectToRequestString(queryParameters)}`)
```

Comment:
I'd prefer to return the fetch result and work with the result where it is called. This keeps this function cleaner. #cohesion

```javascript
return fetch(`${epsgIoBaseUrl}?${UrlUtil.objectToRequestString(queryParameters)}`)
  .then(response => response.json());
```

See below for the changes to `handleSearch`.

Reply:
changed in 3cc756ffa0